py/desitarget/mock/build.py

```diff
@@ -4,1488 +4,439 @@
-        if False:
-            for thisbrick in unique_bricks:
-                brickindx = np.where(brick_info['BRICKNAME'] == thisbrick)[0]
-                if (len(brickindx) != 1):
-                    log.fatal('One or too many matching brick(s) {}! This should not happen...'.format(thisbrick))
-                    ww = np.where( (brick_info['RA'] > source_data['RA'].min()) *
-                                   (brick_info['RA'] < source_data['RA'].max()) *
-                                   (brick_info['DEC'] > source_data['DEC'].min()) *
-                                   (brick_info['DEC'] < source_data['DEC'].max()) )[0]
-                    print(brick_info['BRICKNAME'][ww])
-                    import matplotlib.pyplot as plt
-                    plt.scatter(source_data['RA'], source_data['DEC'])
-                    plt.scatter(brick_info['RA'], brick_info['DEC'])
-                    plt.xlim(source_data['RA'].min()-0.2, source_data['RA'].max()+0.2)
-                    plt.ylim(source_data['DEC'].min()-0.2, source_data['DEC'].max()+0.3)
-                    plt.show()
-                    import pdb ; pdb.set_trace()
-        
-        log.info('Assigned {} {}s to {} unique {}x{} deg2 bricks.'.format(
-            len(brickname), source_name, len(unique_bricks), bricksize, bricksize))
-
-        nbrick = np.zeros((), dtype='i8')
-        t0 = time()
-        def _update_spectra_status(result):
-            if nbrick % 10 == 0 and nbrick > 0:
-                rate = (time() - t0) / nbrick
-                log.info('{} bricks; {:.1f} sec / brick'.format(nbrick, rate))
-            nbrick[...] += 1    # this is an in-place modification
-            return result
-
-        specargs = list()
-        for thisbrick in unique_bricks:
-            specargs.append( (target_name.lower(), mockformat, thisbrick, brick_info,
-                              Spectra, params['dust_dir'], select_targets_function,
-                              source_data, rand, log) )
-
-        if nproc > 1:
-            pool = sharedmem.MapReduce(np=nproc)
-            with pool:
-                out = pool.map(_get_spectra_onebrick, specargs,
-                               reduce=_update_spectra_status)
-        else:
-            out = list()
-            for ii in range(len(unique_bricks)):
-                out.append( _update_spectra_status( _get_spectra_onebrick(specargs[ii]) ) )
+    return log, rand, healpixels, areaperpix
+    
+def _density_fluctuations(params, data, log, nside=16, nside_chunk=128, rand=None):
+    """Density fluctuations model."""
 
-        del source_data # memory clean-up
+    if rand is None:
+        rand = np.random.RandomState()
 
-        # Unpack the results removing any possible bricks without targets.
-        out = list(zip(*out))
+    # Fluctuations model coefficients from --
+    #   https://github.com/desihub/desitarget/blob/master/doc/nb/target-fluctuations.ipynb
+    model = dict()
```

Comment:
What happens for BGS?

Reply:
BGS has density fluctuations built into it, because of the sophistication of the MXXL mock.

Reply:
Also due to imaging?

Reply:
I thought that the density fluctuations model was dependent on the DECaLS results. Are those also included in MXXL?